openqabot/pc_helper.py

```diff
@@ -100,6 +102,133 @@ def apply_publiccloud_pint_image(settings):
     return settings
 
 
+def get_pint_image(name_filter, field, state, query, region=None):
+    """
+    Calculate the most recent image in PINT:
+    1. Compose PINT url using query+state,
+    2. get list of images from the url,
+    3. search the one with name matching the filter,
+    4. return the newest one.
+
+    :param str name_filter passed to get_recent_pint_image
+    :param str field key name in dictionary returned by PINT to be used as ID in the return dictionary
+    :param str state active/inactive used as name of the json file in PINT url
+    :param str query url of PINT, has to have '/' at the end
+    :param str region optional filter, mostly needed by AWS
+    :return Return dictionary has keys NAME and ID
+    """
+    settings = {}
+    try:
+        url = f"{query}{state}.json"
+        log.debug("Pint url:%s name_filter:%s region:%s", url, name_filter, region)
+        images = pint_query(url)["images"]
+        image = get_recent_pint_image(images, name_filter, region=region, state=state)
+        if image is None:
+            raise ValueError(
+                f"Cannot find matching image in PINT with name:[{name_filter}] and state:{state}"
+            )
+        if field not in image.keys() or "name" not in image.keys():
+            raise ValueError(
+                f"Cannot find expected keys '{field}' in the selected image dictionary {image}"
+            )
+        settings["ID"] = image[field]
+        settings["NAME"] = image["name"]
+    except BaseException as e:
+        log.warning(f"get_pint_image handling failed: {e}")
+    log.debug("Return settings:%s", settings)
+    return settings
+
+
+def pint_url(pint_base_url, csp_name):
+    """
+    :param str pint_base_url, usually https://susepubliccloudinfo.suse.com/v1
+    :param str csp_name microsoft, amazon, google, ...
+    :return str PINT url for the images API
+    """
+    url = "/".join([pint_base_url] + [csp_name, "images"]) + "/"
+    log.debug("PINT url:%s", url)
+    return url
+
+
+def sles4sap_pint_gce(name_filter, state, pint_base_url):
+    """
+    Query PINT about GCE images and retrieve the latest one
+    """
+    job_settings = {}
+    ret = get_pint_image(
+        name_filter=name_filter,
+        field="project",
+        state=state,
+        query=pint_url(pint_base_url, "google"),
+    )
+    if any(ret):
+        job_settings["SLES4SAP_QESAP_OS_VER"] = f"{ret['ID']}/{ret['NAME']}"
+        job_settings["SLES4SAP_QESAP_OS_STATE"] = state
```

Comment:
@asmorodskyi yesterday we always talk about the fact that dumping `state` to a setting is not needed, but PC is dumping it too

https://github.com/openSUSE/qem-bot/pull/144/files#diff-5578fec6c7adac2cb35f299569fc8310e4a4e7d1005ac8d1802d3dc75b20ede1R84

seems nothing on the test side it using `PUBLIC_CLOUD_IMAGE_STATE`. Would you like me to remove `PUBLIC_CLOUD_IMAGE_STATE` too?

Reply:
no we exposing it for informational purposes . So for example when you see job for SLE 15 SP5 which running with image in "deprecated" state you suppose to figure out why this is happening because SP5 which is latest currently suppose to have "active" images , so there is issue or with bot or with image publishing pipeline

Reply:
I like the idea, I'll do it for sles4sap and reusing the `PUBLIC_CLOUD_IMAGE_STATE` name